SimulationAgent/SimulationManager.cs

```diff
@@ -228,6 +237,27 @@ public void PrintStats()
                 });
         }
 
+        public async Task SaveStatisticsAsync()
+        {
+            try
+            {
+                var telemetryActors = this.deviceTelemetryActors ?? this.deviceTelemetryActors.Where(a => a.Key.StartsWith(simulation.Id));
```

Comment:
missing `this.` in various places => `this.simulation.Id`

Reply:
fixed